Classes/Controller/BackendServiceController.php

```diff
@@ -182,56 +170,70 @@ public function changeAction(array $changes): void
-        $this->publishingService->publishWorkspace($contentRepository, $workspaceName);
+        try {
+            $workspace = $this->workspaceProvider->provideForWorkspaceName(
+                $command->contentRepositoryId,
+                $command->workspaceName
+            );
+            $workspace->publishChangesInSite($command->siteId);
 
-        $success = new Success();
-        $success->setMessage(sprintf('Published.'));
+            $success = new Success();
+            $success->setMessage(sprintf('Published.'));
 
-        $updateWorkspaceInfo = new UpdateWorkspaceInfo($contentRepositoryId, $workspaceName);
-        $this->feedbackCollection->add($success);
-        $this->feedbackCollection->add($updateWorkspaceInfo);
+            $this->feedbackCollection->add($success);
+        } catch (\Exception $e) {
+            $error = new Error();
+            $error->setMessage($e->getMessage());
+
+            $this->feedbackCollection->add($error);
+        }
         $this->view->assign('value', $this->feedbackCollection);
     }
 
     /**
-     * Publish nodes
+     * Publish all changes in the current document
      *
-     * @psalm-param list<string> $nodeContextPaths
+     * @phpstan-param array<string,string> $command
      */
-    public function publishAction(array $nodeContextPaths, string $targetWorkspaceName): void
+    public function publishChangesInDocumentAction(array $command): void
     {
+        /** @todo send from UI */
+        $contentRepositoryId = SiteDetectionResult::fromRequest($this->request->getHttpRequest())->contentRepositoryId;
+        $command['contentRepositoryId'] = $contentRepositoryId->value;
+        $command['documentId'] = $this->nodeService->deserializeNodeAddress(
+            $command['documentId'],
+            $contentRepositoryId
+        )->nodeAggregateId->value;
+        $command = PublishChangesInDocument::fromArray($command);
+
         $contentRepositoryId = SiteDetectionResult::fromRequest($this->request->getHttpRequest())->contentRepositoryId;
         $contentRepository = $this->contentRepositoryRegistry->get($contentRepositoryId);
-        $nodeAddressFactory = NodeAddressFactory::create($contentRepository);
+        $baseWorkspaceName = $contentRepository->getWorkspaceFinder()->findOneByName(
+            $command->workspaceName
+        )->baseWorkspaceName;
 
         try {
-            $currentAccount = $this->securityContext->getAccount();
-            $workspaceName = WorkspaceNameBuilder::fromAccountIdentifier($currentAccount->getAccountIdentifier());
-
-            $nodeIdentifiersToPublish = [];
-            foreach ($nodeContextPaths as $contextPath) {
-                $nodeAddress = $nodeAddressFactory->createFromUriString($contextPath);
-                $nodeIdentifiersToPublish[] = new NodeIdToPublishOrDiscard(
-                    $nodeAddress->nodeAggregateId,
-                    $nodeAddress->dimensionSpacePoint
-                );
```

Comment:
FYI, as we dont use the node address from the Neos Ui here anymore and their dimension we accidentally trashed partial publishing with respect to dimensions.

But this might be okay as discussed :D